service/src/test/java/bio/terra/pipelines/stairway/imputation/steps/SubmitQuotaConsumedSubmissionStepTest.java

```diff
@@ -0,0 +1,242 @@
+import bio.terra.pipelines.stairway.imputation.ImputationJobMapKeys;
+import bio.terra.pipelines.testutils.BaseEmbeddedDbTest;
+import bio.terra.pipelines.testutils.StairwayTestUtils;
+import bio.terra.pipelines.testutils.TestUtils;
+import bio.terra.rawls.model.MethodConfiguration;
+import bio.terra.rawls.model.MethodRepoMethod;
+import bio.terra.rawls.model.SubmissionReport;
+import bio.terra.rawls.model.SubmissionRequest;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+
+class SubmitQuotaConsumedSubmissionStepTest extends BaseEmbeddedDbTest {
+  @Mock private RawlsService rawlsService;
+  @Captor private ArgumentCaptor<SubmissionRequest> submissionRequestCaptor;
+  @Captor private ArgumentCaptor<MethodConfiguration> updateMethodConfigCaptor;
+  @Captor private ArgumentCaptor<MethodConfiguration> setMethodConfigCaptor;
+  @Mock private SamService samService;
+  @Mock private FlightContext flightContext;
+
+  private final UUID testJobId = TestUtils.TEST_NEW_UUID;
+
+  @BeforeEach
+  void setup() {
+    FlightMap inputParameters = new FlightMap();
+    FlightMap workingMap = new FlightMap();
+
+    when(flightContext.getInputParameters()).thenReturn(inputParameters);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+    when(samService.getTeaspoonsServiceAccountToken()).thenReturn("thisToken");
+  }
+
+  @Test
+  void doStepSuccess() {
+    // setup
+    StairwayTestUtils.constructCreateJobInputs(flightContext.getInputParameters());
+    when(flightContext.getFlightId()).thenReturn(testJobId.toString());
+    MethodConfiguration returnedMethodConfiguration = new MethodConfiguration();
+    when(rawlsService.getCurrentMethodConfigForMethod(
+            "thisToken",
+            TestUtils.CONTROL_WORKSPACE_BILLING_PROJECT,
+            TestUtils.CONTROL_WORKSPACE_NAME,
+            SubmitQuotaConsumedSubmissionStep.QUOTA_CONSUMED_METHOD_NAME))
+        .thenReturn(returnedMethodConfiguration);
+    when(rawlsService.validateMethodConfig(
+            returnedMethodConfiguration,
+            PipelinesEnum.ARRAY_IMPUTATION.getValue(),
+            SubmitQuotaConsumedSubmissionStep.QUOTA_CONSUMED_METHOD_NAME,
+            TestUtils.TEST_PIPELINE_INPUTS_DEFINITION_LIST,
+            TestUtils.TEST_PIPELINE_OUTPUTS_DEFINITION_LIST,
+            TestUtils.TEST_WDL_METHOD_VERSION_1))
+        .thenReturn(true);
+    when(rawlsService.submitWorkflow(
+            eq("thisToken"),
+            submissionRequestCaptor.capture(),
+            eq(TestUtils.CONTROL_WORKSPACE_BILLING_PROJECT),
+            eq(TestUtils.CONTROL_WORKSPACE_NAME)))
+        .thenReturn(new SubmissionReport().submissionId(testJobId.toString()));
```

Comment:
it probably doesn't really matter for the test, but to avoid confusion down the line we should make a new randomUUID (like you do in [PollQuotaConsumedSubmissionStatusStepTest.java](https://github.com/DataBiosphere/terra-scientific-pipelines-service/pull/158/files#diff-695ccf6f54439f1c26fb6048f06bc731db8a1eedebe1a96cc827b34b07c12726)) to use for the submissionId for the quota submission. it's not going to be the same as the flight jobId.

Reply:
great point, updating both tests